library/src/main/java/org/apache/apex/malhar/lib/state/spillable/TimeBasedPriorityQueue.java

```diff
@@ -0,0 +1,128 @@
+package org.apache.apex.malhar.lib.state.spillable;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+public class TimeBasedPriorityQueue<T>
```

Comment:
Have you looked at https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/DelayQueue.html ?